### DISCUSSION TITLE
Bundle condition before launching

### DISCUSSION THREAD

**Discussion by marcopornic** [oleg-shilo/wixsharp#1576] [Q&A]
Hi,

I develop a bundle that Chain an exe and a embedded UI MSI. I need to check some conditions before executing the bundle. I don't really know how to use the Condition param of Bundle:

` 
var bootstrapper =

    new Bundle($"{productName}",
        new ExePackage(@".\cm\CM.exe")
        {
            Name = "ManagedSetup",
            InstallArguments = "/i /ComponentArgs \"*\":\"/qn ADDLOCAL=Complete,EnableNetworkServer\"",
            Compressed = true,
            Permanent = true,
            Vital = true
        },
        new MsiPackage(productMsi) 
        { 
            Compressed = false,
            Vital = true
        }
    )
    {
        Version = project.Version,
        Id = project.Id,
        IconFile = @"logo.ico",
        UpgradeCode = new Guid($"{properties["Guid"]}"),
        Application = application,
        Manufacturer = project.ControlPanelInfo.Manufacturer,
        Condition = "1<0"
    };
`

In this example (Condition = "1<0") when I launch the bundle I have the "**The specified program requires a newer version of windows**" error message displayed. Where can I override condition or message ?

In my case I need to know if Hyper-V feature is enabled on the host (by powershell script or other). If not, the bundle should not be executed.

I'd appreciate some help :)
Greetings
Marc (sorry for my "french english ...")

**Comment by marcopornic** [comment]
Just to clarify, this is "application":

```
WixInternalUIBootstrapperApplication application = new WixInternalUIBootstrapperApplication();
application.LicensePath = @"licence.rtf";
application.LogoFile = @"images\logo.ico";
application.ThemeFile = @"theme\iuipreq.xml";
application.LocalizationFile = @"theme\iuipreq.wxl";
```

I use WixInternalUIBootstrapperApplication because I need Embedded MSI UI

**Reply by oleg-shilo** [reply]
The one that you are using is a native bundle attribute and it does not allow messages.

I am not entirely sure of the solution but according to WiX documentation, it seems like you need a `Bal:Condition` element.
At the moment it's not directly supported and you will need to inject it: 

```c#
bootstrapper.WixSourceGenerated += doc =>
        {
            var bundle = doc.FindSingle("Bundle");
            bundle.AddElement(WixExtension.Bal.ToXName("Condition"), "Condition=1<0; Message=Custom Error Message");
        };
```

**Reply by marcopornic** [reply]
Thanks for your answer.
Can I set a more evolved condition ? Like script or command return for example ?

**Reply by oleg-shilo** [reply]
Unlikely. It has to be an expression that contains WiX properties and literals.
But it is a WiX question so possibly you may find a different answer on Wix forums.

**Reply by oleg-shilo** [reply]
Sorry, I missed the point that BalCondition is already implemented. So you do not need to use XML injection.

YOu can do it normal way:

```c#
bootstrapper.GenericItems.Add(new BalCondition { Condition = "1<0", Message = "Custom Error Message" });

```

**Reply by marcopornic** [reply]
Thanks for your answer ! What kind of conditions are possible ? You said "WiX properties and literals". Is there a list with all WIX properties ? Some examples ?
Thanks !

**Reply by oleg-shilo** [reply]
This is not a WixSharp but a WiX question. You might find it in the WiX documentation.
WixShharp only facilitates the MSI build with the WiX toolset, but it has no control/involvement in WiX schema or runtime behaviour. 
So, probably, WiX forums are a better source for this information.
